compiler/ccgexprs.nim

```diff
@@ -869,16 +871,31 @@ proc genFieldCheck(p: BProc, e: PNode, obj: Rope, field: PSym) =
     v.r.add(".")
     v.r.add(disc.sym.loc.r)
     genInExprAux(p, it, u, v, test)
-    let msg = genFieldDefect(field, disc.sym)
-    let strLit = genStringLiteral(p.module, newStrNode(nkStrLit, msg))
-    if op.magic == mNot:
-      linefmt(p, cpsStmts,
-              "if ($1){ #raiseFieldError($2); $3}$n",
-              [rdLoc(test), strLit, raiseInstr(p)])
-    else:
-      linefmt(p, cpsStmts,
-              "if (!($1)){ #raiseFieldError($2); $3}$n",
-              [rdLoc(test), strLit, raiseInstr(p)])
+    var msg = toFileLineCol(p.config, e.info)
+      # shows the most relevant lineinfo even with -d:release;
+      # should have no runtime cost (unlike -d:stacktrace)
+      # this could be added to all/most chcks.nim errors, and perhaps
+      # can be disabled with a compile flag.
```

Comment:
This should respect the normal stacktrace configuration options.
Introducing an option to show locations of errors without a stacktrace can be added in a seperate PR.

Reply:
`--stacktrace` is unrelated and incurs runtime costs; lots of error messages show location info (eg assert errors)